arduino/src/main.cpp

```diff
@@ -20,40 +24,92 @@ void setup(){
     while(WiFi.status() != WL_CONNECTED){
         Serial.print(".");
         delay(100);
+        
     }
-
-    
+    pinMode(LED,OUTPUT);
+    pinMode(BUTTON,INPUT_PULLUP);
 }
-
-void loop() {
-    // Odczytaj stan GPIO
-   int gpioState = 10 ;
+int send_status_request(int buttonState) 
+{
+    
     DynamicJsonDocument jsonDoc(200);
-    jsonDoc["gpioState"] = gpioState;
+    jsonDoc["place"] =  MIEJSCE;
+    if(buttonState==LOW)
+    {
+        jsonDoc["state"] = "hanged";
+        delay(30000);
+    }
+    else
+    {
+        jsonDoc["state"]="empty";
+        delay(30000);
+    }
+    jsonDoc["board_id"] = BOARD_ID;
     String payload;
-    serializeJson(jsonDoc,payload);
+    serializeJson(jsonDoc, payload);
     // Wyślij żądanie POST na serwer Flask
     HTTPClient http;
-    http.begin(serverAddress);
+    http.begin(SERVER_ADDRESS);
     http.addHeader("Content-Type", "application/json");
-     int httpResponseCode = http.sendRequest("GET", payload);
+    int httpResponseCode = http.sendRequest("POST", payload);
+    http.end();
+    return httpResponseCode;
+}
+
+
+void loop() 
+{   
+    int buttonState = digitalRead(BUTTON);
+    if(buttonState==LOW)   
+    {
+        digitalWrite(LED,HIGH);
+    }
+    else
+    {
+        digitalWrite(LED,LOW);
+    }
+    HTTPClient http;
+    http.begin(SERVER_ADDRESS);
+    int httpResponseCode = send_status_request(buttonState);
+    Serial.print(httpResponseCode);
     if(httpResponseCode > 0)
     {
+        
         Serial.printf("HTTP Response code: %d\n",httpResponseCode );
         String response = http.getString();
+        Serial.print(MIEJSCE);
         DynamicJsonDocument jsonDoc(200);
         deserializeJson(jsonDoc, response);
         String success = jsonDoc["success"].as<String>();
         Serial.println(response);

```

Comment:
Po co jest ten kod?

Skoro już wysłałeś swój POSTem to chyba już nie ma potrzeby, aby odczytywać zawartości jsona odpowiedzi.

Poza tym status (sukces porażka etc.) wynika z odpowiedniego numeru HTTP https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
Nie musisz czytać, czy w środku jest słowo success.